scripts/zones/Aydeewa_Subterrane/npcs/qm2.lua

```diff
@@ -7,13 +7,12 @@ local ID = require("scripts/zones/Aydeewa_Subterrane/IDs")
 require("scripts/globals/npc_util")
 -----------------------------------
 
-function onTrade(player,npc,trade)
-    if npcUtil.tradeHas(trade, 2572) and not GetMobByID(ID.mob.PANDEMONIUM_WARDEN):isSpawned() then -- Pandemonium Key
+function onTrade(player, npc, trade)
+    if npcUtil.tradeHas(trade, 2572) and npcUtil.popFromQM(player, npc, ID.mob.PANDEMONIUM_WARDEN) then -- Pandemonium Key
```

Comment:
We might run into problems if Pandemonium Warden uses multiple IDs during the fight.

Not an issue to be solved now, but "keep in mind for later".